src/spaceone/repository/service/plugin_service.py

```diff
@@ -286,17 +287,33 @@ def list(self, params):
         """
 
         repo_mgr: RepositoryManager = self.locator.get_manager('RepositoryManager')
-        repository_id = params['repository_id']
-        repo_vo = repo_mgr.get_repository(repository_id)
-
-        plugin_mgr = self._get_plugin_manager_by_repo(repo_vo)
         query = params.get('query', {})
         only = query.get('only', [])
 
         if 'registry_url' in only:
             only.remove('registry_url')
 
-        return plugin_mgr.list_plugins(query)
+        if params.get('repository_id'):
```

Comment:
It would be better to declare the `repository_id` first and then compare the conditions.
Or use [Assignment Expression](https://peps.python.org/pep-0572/), which is supported since python 3.8.

`if repository_id := params.get('repository_id'):`